pkg/controllers/workgenerator/controller.go

```diff
@@ -220,6 +221,38 @@ func (r *Reconciler) Reconcile(ctx context.Context, req controllerruntime.Reques
 	return controllerruntime.Result{}, syncErr
 }
 
+// updateBindingStatusWIthRetry sends the update request to API server with retry.
+func (r *Reconciler) updateBindingStatusWithRetry(ctx context.Context, resourceBinding *fleetv1beta1.ClusterResourceBinding) error {
+	// Retry only for specific errors or conditions
+	err := r.Client.Status().Update(ctx, resourceBinding)
+	if err != nil {
+		klog.ErrorS(err, "Failed to update the resourceBinding status, will retry", "resourceBinding", klog.KObj(resourceBinding), "resourceBindingStatus", resourceBinding.Status)
+		errAfterRetries := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+			var latestBinding fleetv1beta1.ClusterResourceBinding
+			if err := r.Client.Get(ctx, client.ObjectKeyFromObject(resourceBinding), &latestBinding); err != nil {
+				return err
+			}
+			// Work generator is the only controller that updates conditions excluding rollout started which is updated by rollout controller.
+			if rolloutCond := latestBinding.GetCondition(string(fleetv1beta1.ResourceBindingRolloutStarted)); rolloutCond != nil && resourceBinding.GetCondition(string(fleetv1beta1.ResourceBindingRolloutStarted)) == nil {
```

Comment:
what happens when the "resourceBinding.GetCondition(string(fleetv1beta1.ResourceBindingRolloutStarted)) != nil"

Reply:
It’s added twice

Reply:
During the first retry it doesn’t exist so it gets added. If it retries again it append another rollout started condition, but realized that it could be an updated rollout started condition when it retries.